### PR TITLE
3dfsc and cryoEF

### PR DESCRIPTION
**Pull request by azazellochg**
This PR adds two new programs and tests for them: 

- https://github.com/nysbc/Anisotropy (3D FSC)
- https://www.mrc-lmb.cam.ac.uk/crusso/cryoEF/

Additionally, it adds a neededProgs kwarg to addPackage function in install/funcs.py

The remaining piece that I need help with is [here](https://github.com/I2PC/scipion/compare/devel...azazellochg:az-3dfsc#diff-733745b6908c2853b837d7b6413ba8b1R522): commands kwarg in addPackage only checks if target file exists. However I cannot predict where the target will be, because conda virtualenv folder can be anywhere defined by user... Any ideas?

### CODE REVIEW

File: install/script.py [hunk 1]
```diff
@@ -527,4 +527,16 @@
                tar='ethan-1.2.tgz',
                commands=[('make', 'ethan')])
 
+fsc_commands = [('conda env create -f environment.yml',
+                 '/home/gsharov/soft/EMAN2.2/envs/fsc')]
```
Comment (pconesa) [inline install/script.py]: I understand the "conda" thing...but what is the other path and, can it be done in a generic way? (avoiding /home/gsharov)

Comment (azazellochg) [reply install/script.py]: @pconesa that's exactly what I asked about in the PR :)

Comment (azazellochg) [reply install/script.py]: the ideal fix will be:
conda env list | grep -q fsc -> returns 0, otherwise run 'conda env create -f environment.yml'

Comment (pconesa) [reply install/script.py]: @azazellochg , is the "fsc" needed for installation/compilation or just to run.

Comment (azazellochg) [reply install/script.py]: @pconesa yes, the fsc is a name for a folder containing the Anaconda virtualenv

Comment (pconesa) [reply install/script.py]: I'm not familiar either...and to run it....do you need to activate the conda virtual env?

**Comment by azazellochg** [comment]
Yes, it is activated inside the protocol

On Jan 17, 2018 19:08, "Pablo Conesa" <notifications@github.com> wrote:

*@pconesa* commented on this pull request.
------------------------------

In install/script.py
<https://github.com/I2PC/scipion/pull/1428#discussion_r162149171>:

> @@ -527,4 +527,16 @@
                tar='ethan-1.2.tgz',
                commands=[('make', 'ethan')])

+fsc_commands = [('conda env create -f environment.yml',
+                 '/home/gsharov/soft/EMAN2.2/envs/fsc')]

I'm not familiar either...and to run it....do you need to activate the
conda virtual env?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/I2PC/scipion/pull/1428#discussion_r162149171>, or mute
the thread

File: install/script.py (hunk 1 above)
Comment (azazellochg) [reply install/script.py]: @pconesa Yes, it is activated inside the protocol

**Comment by azazellochg** [comment]
- [ ] todo: move tests to package folders

**Comment by azazellochg** [comment]
@pconesa I have fixed the conda thing during install, so you can review this PR.

**Comment by pconesa** [comment]
Nice! Good job @azazellochg

**Pull request merged**